### ISSUE TITLE
About using the CuraEngine execution file.

### ISSUE THREAD

**Issue by YtcBrian** [Ultimaker/CuraEngine#1170]
Hi, 

I am trying to communicate to the CuraEngine.exe with my custom Application via the command.
And i had already build out the CuraEngine.exe.  
The question is : Is there any limit of license issue for using the CuraEngine.exe with custom application(Commercial, open source...etc)?

**Comment by Ghostkeeper** [comment]
I'm not a lawyer so you need to take my interpretation with a grain of salt. CuraEngine uses the AGPLv3 license, which allows you to use the software as a dependency as long as your software is compatible with GPL as well (and "compatible with GPL" is a whole topic on its own too but generally it means that your application must be open source). You are not allowed to use CuraEngine as a dependency if you are not compatible with GPL. In layman's terms, closed source software is not allowed to use it as a dependency.

However "using it as a dependency" is contested territory. From what I've seen, most people seem to think that providing AGPL software as an optional component is okay, like a plug-in. Most people seem to assume that shipping AGPL software with the default installation is not allowed. We need to be careful with saying anything about this as people are just [repeating what others say](https://softwareengineering.stackexchange.com/questions/107883/agpl-what-you-can-do-and-what-you-cant) and there is some Chinese Whispering going on then.
